src/app/public/modules/split-view/split-view-adapter.service.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  ElementRef,
+  Injectable,
+  Renderer2,
+  RendererFactory2
+} from '@angular/core';
+
+import {
+  MutationObserverService,
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import {
+  fromEvent,
+  Subject
+} from 'rxjs';
+
+import {
+  take,
+  takeUntil
+} from 'rxjs/operators';
+
+@Injectable()
+export class SkySplitViewAdapterService {
+
+  private observer: MutationObserver;
+
+  private renderer: Renderer2;
+
+  constructor(
+    private observerService: MutationObserverService,
+    private rendererFactory: RendererFactory2,
+    private windowRef: SkyAppWindowRef
+  ) {
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
+  }
+
+  public bindHeightToWindow(elementRef: ElementRef, unsubscribeSubject: Subject<void>) {
+    if (elementRef.nativeElement.offsetParent === document.body) {
+      this.observer = this.observerService.create((mutations: MutationRecord[]) => {
+        this.setSplitViewBoundHeights(elementRef);
+      });
+
+      fromEvent(this.windowRef.nativeWindow, 'resize')
+        .pipe(
+          takeUntil(unsubscribeSubject)
+        ).subscribe(() => {
+          this.setSplitViewBoundHeights(elementRef);
+        });
+
+      const config = { attributes: true, attributeFilter: ['style'], childList: false, characterDate: false };
+
+      // This observer makes sure that the split view reacts if the body styling is change but thing
+      // such as the action bar.
+      this.observer.observe(document.body, config);
+      this.setSplitViewBoundHeights(elementRef);
+
+      unsubscribeSubject
+        .pipe(take(1))
+        .subscribe(() => {
+          this.observer.disconnect();
+          const splitViewElement = elementRef.nativeElement.querySelector('.sky-split-view');
+          this.renderer.removeStyle(splitViewElement, 'max-height');
+          this.renderer.removeStyle(splitViewElement, 'min-height');
+        });
+    }
+  }
+
+  private setSplitViewBoundHeights(elementRef: ElementRef) {
+    const splitViewElement = elementRef.nativeElement.querySelector('.sky-split-view');
```

Comment:
You grab `.sky-split-view` a couple times in here. Should we make it more globally accessible to the whole service?

Reply:
We can't unless we require another function call into the adapter to set this value. We use the `elementRef` that is passed into both functions to grab the element and that isn't something we have during the service's construction.